server/routes/taskList.ts

```diff
@@ -128,23 +127,9 @@ export = (
           postApproval: licenceStatus.postApproval,
         }
 
-        if (
-          req.user.role === 'CA' &&
-          licenceStatus.stage === 'ELIGIBILITY' &&
-          licenceStatus.tasks.eligibility === 'DONE'
-        ) {
-          const errorCode = await caService.getReasonForNotContinuing(bookingId, res.locals.token)
-
-          if (errorCode) {
-            return res.render('taskList/taskListBuilder', {
-              ...model,
-              taskListModel: getTasksForBlocked(errorCode),
-              errors: [],
-            })
-          }
-        }
+        const errorCode = await caService.getReasonForNotContinuing(bookingId, res.locals.token)
```

Comment:
So this would make an extra api call for every tasklist where in quite a lot of cases it'd be irrelevant.. in other languages we could make this lazy and pass this through - only calling it when necessary but in JS using async await it means we'd have to change the [colour of the function](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/#:~:text=Red%20functions%20are%20asynchronous%20ones,just%20made%20a%20red%20function.) we're calling to also it to do async await also. 

One way around this could be to split out a getCaTasklist in taskListModels,
so here we:
```js
if (role === 'CA') {
  const errorCode = await caService.getReasonForNotContinuing(bookingId, res.locals.token)
  return res.render('taskList/taskListBuilder', {
              ...model,
              taskListModel: getCaTaskList(licenceStatus, errorCode, licence || {}),
              errors: [],
            })
}

const taskListModel = getTaskListModel(req.user.role, postRelease, licenceStatus, licence || {})

        return res.render('taskList/taskListBuilder', {
          ...model,
          taskListModel,
          errors: [],
        })
```